tests/xharness/Jenkins/Jenkins.cs

```diff
@@ -554,7 +561,13 @@ async Task<IEnumerable<TestTask>> CreateRunSimulatorTasksAsync ()
 			}
 
 			var testVariations = CreateTestVariations (runSimulatorTasks, (buildTask, test, candidates) =>
-				new RunSimulatorTask (simulators, buildTask, processManager, candidates?.Cast<SimulatorDevice> () ?? test.Candidates)).ToList ();
+				new RunSimulatorTask (
+					jenkins: this,
+					simulators: simulators,
+					buildTask: buildTask,
+					processManager: processManager,
+					tunnelBore: tunnelBore,
+					candidates: candidates?.Cast<SimulatorDevice> () ?? test.Candidates)).ToList ();
```

Comment:
A little confused about `candidates` here. Why do we need to switch between these two different values?